Makefile

```diff
@@ -4,26 +4,39 @@ eslint:
 stylelint:
 	npm run stylelint
 
+build-test:
+	npm install
+	npm run build
+
 unit-test:
 	npm run test:cov
 
 end-to-end-test:
-	git clone https://github.com/weldr/welder-deployment
-	make -C welder-deployment/ weld-f25
-
-	sudo docker build -f ./test/end-to-end/Dockerfile -t weld/end-to-end:latest ./test/end-to-end/
-
-	git clone ../welder-web welder-deployment/welder-web
-	make -C welder-deployment/ repos
-	
-	if [ ! -d ./welder-deployment/mddb ]; then \
-		mkdir ./welder-deployment/mddb; \
-	fi; \
-	sudo docker-compose -f welder-deployment/docker-compose.yml build
-	wget https://s3.amazonaws.com/weldr/metadata.db -O welder-deployment/mddb/metadata.db
-	sudo docker-compose -f welder-deployment/docker-compose.yml -p welder up -d
-
-	sudo docker run --rm --name welder_end_to_end --network welder_default \
-	    -v test-result-volume:/result -v `pwd`/welder-deployment/mddb:/mddb -e MDDB='/mddb/metadata.db' \
+	sudo docker run --rm --name end_to_end --network welder \
+	    -v test-result-volume:/result -v `pwd`:/mddb -e MDDB='/mddb/metadata.db' \
 	    weld/end-to-end:latest \
 	    xvfb-run -a -s '-screen 0 1024x768x24' npm run test
+
+e2e-before-install:
+	if [ -f ${CACHE_FILE_FEDORA} ]; then gunzip -c ${CACHE_FILE_FEDORA} | sudo docker load; fi
+	if [ -f ${CACHE_FILE_WEB} ]; then gunzip -c ${CACHE_FILE_WEB} | sudo docker load; fi
+	if [ -f ${CACHE_FILE_API} ]; then gunzip -c ${CACHE_FILE_API} | sudo docker load; fi
+	if [ -f ${CACHE_FILE_END_TO_END} ]; then gunzip -c ${CACHE_FILE_END_TO_END} | sudo docker load; fi
+	git clone https://github.com/weldr/welder-deployment.git
+	git clone https://github.com/weldr/bdcs-api-rs.git
+	wget https://s3.amazonaws.com/weldr/metadata.db
+	sudo docker network create welder
+	mv welder-deployment/Dockerfile-weld-f25 welder-deployment/Dockerfile
```

Comment:
I feel like the above 5 lines should go into an else block somewhere. Probably for the if on line 23.

Tip if you want to write a multi line script in Makefile then use \ and ; characters.